File: tvr-core/src/main/java/com/willcurrie/hex/WhitespaceElement.java
package com.willcurrie.hex;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;
import org.apache.commons.lang.builder.ToStringBuilder;
import org.apache.commons.lang.builder.ToStringStyle;

public class WhitespaceElement implements HexDumpElement{
    
    private final String value;

    public WhitespaceElement(String value) {
        this.value = value;
    }

    @Override
    public boolean isByte() {
        return false;
    }

    public String getValue() {
        return value;
    }

    @Override
    public int hashCode() {
        return HashCodeBuilder.reflectionHashCode(this);
    }

    @Override
    public boolean equals(Object obj) {
        return EqualsBuilder.reflectionEquals(this, obj);
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: src/main/java/io/github/binaryfoo/hex/WhitespaceElement.kt
package io.github.binaryfoo.hex

import org.apache.commons.lang.builder.EqualsBuilder
import org.apache.commons.lang.builder.HashCodeBuilder
import org.apache.commons.lang.builder.ToStringBuilder
import org.apache.commons.lang.builder.ToStringStyle

public data class WhitespaceElement(val value: String) : HexDumpElement {

    override fun isByte(): Boolean {
        return false
    }
}
